python/pyspark/sql/tests.py

```diff
@@ -1947,6 +1968,29 @@ def test_sparksession_with_stopped_sparkcontext(self):
         df.collect()
 
 
+class UDFInitializationTests(unittest.TestCase):
```

Comment:
And added a separate test case checking `SparkContext` and `SparkSession` state.